ext/opcache/jit/zend_jit_helpers.c

```diff
@@ -1049,7 +1034,7 @@ static zend_never_inline zend_long zend_check_string_offset(zval *dim/*, int typ
 				}
 				return offset;
 			}
-			zend_jit_illegal_string_offset(dim);
+			zend_illegal_container_offset("string", dim, BP_VAR_R);
```

Comment:
`zend_check_string_offset` (in zend_jit_helpers.c) can be called for R or W.

Reply:
Decided to make it pass the type.